addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/SeleniumScript.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.selenium;
+
+import org.openqa.selenium.WebDriver;
+
+/** A script run whenever a browser is launched using selenium. */
+public interface SeleniumScript {
+
+    void browserLaunched(
```

Comment:
IMO we should use a parameter object (or add a "utils" param) to allow to add more state/behaviour without breaking compatibility.

Reply:
Yeah, good thinking - done.